pkg/lib/userimport/service.go

```diff
@@ -173,6 +173,9 @@ func (s *UserImportService) importRecordInConn(
 	if record != nil {
 		record.Redact()
 		detail.Record = record
+	} else {
+		// record is nil means parsing is failed, return the raw input
+		detail.Record = rawMessage
```

Comment:
It is no good to include the raw message because it could contain un-redacted information. Drop this commit. Having the correct index should be enough to pinpoint the problematic row.

Reply:
As our api has a body size limit, importers could split an import into multiple jobs. If we only return an index, it requires additional calculation / mapping to understand the error. It will be much easier to understand the error if we include the input object in the result. If your concern is un-redacted password / mfa data, I prefer to redact the raw input and still include it in the result.

Reply:
I've added a commit to redact password and mfa from user input, see if it is ok?